### PR TITLE
fix(http-server): limit threads per service

### PR DESCRIPTION
**Pull request by bojiang**

### CODE REVIEW

File: src/_bentoml_impl/server/app.py [hunk 1]
```diff
@@ -8,6 +8,7 @@
 import typing as t
 from pathlib import Path
 
+import anyio
```
Comment (aarnphm) [inline src/_bentoml_impl/server/app.py:11]: Why do we decide to only limit to one thread per API function?

File: src/_bentoml_impl/server/app.py [hunk 2]
```diff
@@ -36,6 +37,31 @@
 R = t.TypeVar("R")
 
 
+LIMITER_MAP: dict[str, t.Any] = {}
```
Comment (frostming) [inline src/_bentoml_impl/server/app.py:40]: It's suboptimal to use a global map, instead I would prefer to store this map inside the app factory.

File: src/_bentoml_impl/server/app.py (hunk 1 above)
Comment (bojiang) [reply src/_bentoml_impl/server/app.py:11]: answered offline

File: src/_bentoml_impl/server/app.py (hunk 2 above)
Comment (bojiang) [reply src/_bentoml_impl/server/app.py:40]: makes more sense since we need service level

File: src/_bentoml_sdk/service/config.py [hunk 3]
```diff
@@ -194,6 +194,7 @@ class ServiceConfig(TypedDict, total=False):
     max_runner_connections: Posint
     resources: ResourceSchema
     workers: WorkerSchema
+    threads: Posint
```
Comment (frostming) [inline src/_bentoml_sdk/service/config.py:197]: Also need to update the schema in `configurations/v2`

**Review by aarnphm — APPROVED** [review]

**Pull request merged**